src/catalog/catalog.cpp

```diff
@@ -573,15 +589,15 @@ std::shared_ptr<const storage::Layout> Catalog::CreateLayout(
   // Ensure that the new layout
   PELOTON_ASSERT(layout_oid < INVALID_OID);
   auto new_layout = std::shared_ptr<const storage::Layout>(
-      new const storage::Layout(column_map, layout_oid));
+      new const storage::Layout(column_map, column_map.size(), layout_oid));
```

Comment:
Because there are some cases that a size of `column_map` doesn't equal to `num_columns`. This occurs when a layout type is row store or column store. In this case, `column_map` is empty, even when `num_columns` is set to a correct number rather than 0. This problem occurs in case of recovering layout object from layout catalog.